include/boost/asio/windows_sspi/stream.hpp

```diff
@@ -92,10 +98,51 @@ class sspi_impl {
   CtxtHandle* m_context;
 };
 
+// TODO: Move away from this file
+template <typename NextLayer, typename ConstBufferSequence>
+struct async_write_impl {
+  async_write_impl(NextLayer& next_layer, const ConstBufferSequence& buffer, std::shared_ptr<sspi_impl> sspi)
+    : m_next_layer(next_layer)
+    , m_buffer(buffer)
+    , m_sspi(sspi)
```

Comment:
`std::move(sspi)` to avoid an unnecessary atomic operation

Reply:
Good point. Thanks.